plugins/steelseries/fu-steelseries-sonic.c

```diff
@@ -247,7 +253,10 @@ fu_steelseries_sonic_read_from_flash(FuDevice *device,
 					    0x0,
 					    STEELSERIES_BUFFER_FLASH_TRANSFER_SIZE);
 	fu_progress_set_id(progress, G_STRLOC);
-	fu_progress_set_status(progress, FWUPD_STATUS_DEVICE_READ);
+	if (verify)
+		fu_progress_set_status(progress, FWUPD_STATUS_DEVICE_VERIFY);
+	else
+		fu_progress_set_status(progress, FWUPD_STATUS_DEVICE_READ);
```

Comment:
rather than this, could you set the status on the "parent" FuProgress?

Reply:
Could we imaging that the child inherits from the parent value at creation?

Reply:
what is the function to get parent?

Reply:
Nah, I mean set the progress *before* calling fu_steelseries_sonic_read_from_flash

Reply:
Hum, I need some time to convince myselt it is the way to go.

For now, my brain is blocking on doing:

```
fu_progress_set_id(progress, G_STRLOC);
fu_progress_set_status(progress, ...);
fu_progress_set_steps(progress, chunks->len);
```
where `...` is either `fu_progress_get_status(fu_progress_get_parent(progress)));` or `fu_progress_get_parent_status(progress)`.

so the code is following "almost" the same pattern.

Or, not having the `fu_progress_set_status(progress, ...);` because it is kind of inherited from its parent at creation.

Reply:
Cool

Reply:
[bff9c5b](https://github.com/fwupd/fwupd/pull/4669/commits/bff9c5b3188994525e2da478da9ca85e7085bf5a) does the trick...

I have drop the `READ`/`VERIFY` status, and I can go even more further! `ERASE/WRITE`. to be continued

Reply:
I could save one `_set _status(..., WRITE)` in the end.